doc/man3/SSL_SESSION_get0_hostname.pod

```diff
@@ -24,7 +24,10 @@ SSL_SESSION_set1_alpn_selected
 =head1 DESCRIPTION
 
 SSL_SESSION_get0_hostname() retrieves the SNI value that was sent by the
-client when the session was created, or NULL if no value was sent.
+client when the session was created if it was accepted by the server and TLSv1.2
+or below was negotiated. Otherwise NULL is returned. Note that in TLSv1.3 the
+SNI hostname is negotiated with each handshake including resumption handshakes
+and is therefore never associated with the session.
```

Comment:
Oops, I think I forgot about this bit during our previous discussion about (not) copying the negotiated name into the session for TLS 1.3 -- this text pretty strongly suggests that ```SSL_SESSION_get0_hostname()``` will always return NULL for TLS 1.3 sessions, but I don't think that's the actual behavior with this PR.